### PR TITLE
CATEGORY: added new category entity

### PR DESCRIPTION
**Pull request by kenu21**

### CODE REVIEW

File: src/main/java/com/keniu/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -27,4 +28,5 @@ public class CreateBookRequestDto {
     private String description;
     @Size(max = 255)
     private String coverImage;
+    private Set<Long> categoryIds;
```
Comment (Elena-Bruyako) [inline src/main/java/com/keniu/dto/CreateBookRequestDto.java:31]: Do we need validate categoryIds?

File: src/main/java/com/keniu/dto/CreateCategoryRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+
+@Data
+public class CreateCategoryRequestDto {
+
+    @NotBlank
+    @Size(max = 255)
+    @Column(nullable = false)
```
Comment (Elena-Bruyako) [inline src/main/java/com/keniu/dto/CreateCategoryRequestDto.java]:
```suggestion
```

File: src/main/java/com/keniu/models/Book.java [hunk 3]
```diff
@@ -57,6 +62,18 @@ public class Book {
     /** A brief description of the book. */
     private String description;
 
+    /**
+     * A set of categories that the book belongs to.
+     * Each book can have multiple categories, such as genre, author-related categories, etc.
+     */
+    @ManyToMany
+    @JoinTable(
```
Comment (Elena-Bruyako) [inline src/main/java/com/keniu/models/Book.java:70]:
```suggestion
    @JoinTable(name = "books_categories",
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/keniu/dto/CreateCategoryRequestDto.java (hunk 2 above)
Comment (kenu21) [reply src/main/java/com/keniu/dto/CreateCategoryRequestDto.java:13]: From the task:
name (String, not null)

File: src/main/java/com/keniu/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (kenu21) [reply src/main/java/com/keniu/dto/CreateBookRequestDto.java:31]: What do you suggest?

File: src/main/java/com/keniu/dto/CreateCategoryRequestDto.java (hunk 2 above)
Comment (kenu21) [reply src/main/java/com/keniu/dto/CreateCategoryRequestDto.java]: Removed

File: src/main/java/com/keniu/models/Book.java (hunk 3 above)
Comment (kenu21) [reply src/main/java/com/keniu/models/Book.java:70]: Added

File: src/main/java/com/keniu/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (Elena-Bruyako) [reply src/main/java/com/keniu/dto/CreateBookRequestDto.java:31]: You can use `@NotEmpty` and `@Positive` for Long

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
added suggestion

File: src/main/java/com/keniu/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (kenu21) [reply src/main/java/com/keniu/dto/CreateBookRequestDto.java:31]: book may not have category, so I think it's better to leave it as it is

Comment (Elena-Bruyako) [reply src/main/java/com/keniu/dto/CreateBookRequestDto.java:31]: Deal)

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**